cpp/src/arrow/array/array_binary.cc

```diff
@@ -125,12 +125,8 @@ FixedSizeBinaryArray::FixedSizeBinaryArray(const std::shared_ptr<DataType>& type
                                            int64_t length,
                                            const std::shared_ptr<Buffer>& data,
                                            const std::shared_ptr<Buffer>& null_bitmap,
-                                           int64_t null_count, int64_t offset)
-    : PrimitiveArray(type, length, data, null_bitmap, null_count, offset),
-      byte_width_(checked_cast<const FixedSizeBinaryType&>(*type).byte_width()) {}
```

Comment:
No, because the added `SetData` + `ArrayData::Make` call does the heavy lifting.
(yes, our parallel hierarchy of constructors + `SetData` is confusing, it took me some time to get this PR right)